src/actions/attack.cpp

```diff
@@ -129,13 +129,16 @@ battle_context_unit_stats::battle_context_unit_stats(const unit& u,
 		opp_ctx.emplace(opp_weapon->specials_context(&opp, &u, opp_loc, u_loc, !attacking, weapon));
 	}
 
-	slows = weapon->get_special_bool("slow");
-	drains = !opp.get_state("undrainable") && weapon->get_special_bool("drains");
-	petrifies = weapon->get_special_bool("petrifies");
-	poisons = !opp.get_state("unpoisonable") && weapon->get_special_bool("poison") && !opp.get_state(unit::STATE_POISONED);
+	slows = weapon->get_special_bool("slow")|| under_leadership("slow", units, u_loc, attacking, 0, weapon, opp_weapon).second;
```

Comment:
I'd prefer if all these `under_leadership` calls were actually called from within `get_special_bool` or `get_special`, rather than here where the value is needed.

To put it another way, `get_special_bool("slow")` should return true if the attack should, for any reason, have the slow special, even if it's due to some adjacent unit's ability rather than the unit's innate ability.